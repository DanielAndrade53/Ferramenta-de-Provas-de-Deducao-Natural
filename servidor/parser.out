Created by PLY version 3.11 (http://www.dabeaz.com/ply)

Unused terminals:

    LPAREN
    RPAREN

Grammar

Rule 0     S' -> Program
Rule 1     Program -> ExpressionList
Rule 2     ExpressionList -> <empty>
Rule 3     ExpressionList -> Expression
Rule 4     ExpressionList -> Expression ExpressionList
Rule 5     Expression -> ID
Rule 6     Expression -> Expression BinaryOp Expression
Rule 7     BinaryOp -> ARROW
Rule 8     BinaryOp -> CONJ
Rule 9     BinaryOp -> DISJ
Rule 10    BinaryOp -> NOT

Terminals, with rules where they appear

ARROW                : 7
CONJ                 : 8
DISJ                 : 9
ID                   : 5
LPAREN               : 
NOT                  : 10
RPAREN               : 
error                : 

Nonterminals, with rules where they appear

BinaryOp             : 6
Expression           : 3 4 6 6
ExpressionList       : 1 4
Program              : 0

Parsing method: LALR

state 0

    (0) S' -> . Program
    (1) Program -> . ExpressionList
    (2) ExpressionList -> .
    (3) ExpressionList -> . Expression
    (4) ExpressionList -> . Expression ExpressionList
    (5) Expression -> . ID
    (6) Expression -> . Expression BinaryOp Expression

    $end            reduce using rule 2 (ExpressionList -> .)
    ID              shift and go to state 4

    Program                        shift and go to state 1
    ExpressionList                 shift and go to state 2
    Expression                     shift and go to state 3

state 1

    (0) S' -> Program .



state 2

    (1) Program -> ExpressionList .

    $end            reduce using rule 1 (Program -> ExpressionList .)


state 3

    (3) ExpressionList -> Expression .
    (4) ExpressionList -> Expression . ExpressionList
    (6) Expression -> Expression . BinaryOp Expression
    (2) ExpressionList -> .
    (3) ExpressionList -> . Expression
    (4) ExpressionList -> . Expression ExpressionList
    (7) BinaryOp -> . ARROW
    (8) BinaryOp -> . CONJ
    (9) BinaryOp -> . DISJ
    (10) BinaryOp -> . NOT
    (5) Expression -> . ID
    (6) Expression -> . Expression BinaryOp Expression

  ! reduce/reduce conflict for $end resolved using rule 2 (ExpressionList -> .)
    $end            reduce using rule 2 (ExpressionList -> .)
    ARROW           shift and go to state 7
    CONJ            shift and go to state 8
    DISJ            shift and go to state 9
    NOT             shift and go to state 10
    ID              shift and go to state 4

  ! $end            [ reduce using rule 3 (ExpressionList -> Expression .) ]

    Expression                     shift and go to state 3
    ExpressionList                 shift and go to state 5
    BinaryOp                       shift and go to state 6

state 4

    (5) Expression -> ID .

    ARROW           reduce using rule 5 (Expression -> ID .)
    CONJ            reduce using rule 5 (Expression -> ID .)
    DISJ            reduce using rule 5 (Expression -> ID .)
    NOT             reduce using rule 5 (Expression -> ID .)
    ID              reduce using rule 5 (Expression -> ID .)
    $end            reduce using rule 5 (Expression -> ID .)


state 5

    (4) ExpressionList -> Expression ExpressionList .

    $end            reduce using rule 4 (ExpressionList -> Expression ExpressionList .)


state 6

    (6) Expression -> Expression BinaryOp . Expression
    (5) Expression -> . ID
    (6) Expression -> . Expression BinaryOp Expression

    ID              shift and go to state 4

    Expression                     shift and go to state 11

state 7

    (7) BinaryOp -> ARROW .

    ID              reduce using rule 7 (BinaryOp -> ARROW .)


state 8

    (8) BinaryOp -> CONJ .

    ID              reduce using rule 8 (BinaryOp -> CONJ .)


state 9

    (9) BinaryOp -> DISJ .

    ID              reduce using rule 9 (BinaryOp -> DISJ .)


state 10

    (10) BinaryOp -> NOT .

    ID              reduce using rule 10 (BinaryOp -> NOT .)


state 11

    (6) Expression -> Expression BinaryOp Expression .
    (6) Expression -> Expression . BinaryOp Expression
    (7) BinaryOp -> . ARROW
    (8) BinaryOp -> . CONJ
    (9) BinaryOp -> . DISJ
    (10) BinaryOp -> . NOT

  ! shift/reduce conflict for ARROW resolved as shift
  ! shift/reduce conflict for CONJ resolved as shift
  ! shift/reduce conflict for DISJ resolved as shift
  ! shift/reduce conflict for NOT resolved as shift
    ID              reduce using rule 6 (Expression -> Expression BinaryOp Expression .)
    $end            reduce using rule 6 (Expression -> Expression BinaryOp Expression .)
    ARROW           shift and go to state 7
    CONJ            shift and go to state 8
    DISJ            shift and go to state 9
    NOT             shift and go to state 10

  ! ARROW           [ reduce using rule 6 (Expression -> Expression BinaryOp Expression .) ]
  ! CONJ            [ reduce using rule 6 (Expression -> Expression BinaryOp Expression .) ]
  ! DISJ            [ reduce using rule 6 (Expression -> Expression BinaryOp Expression .) ]
  ! NOT             [ reduce using rule 6 (Expression -> Expression BinaryOp Expression .) ]

    BinaryOp                       shift and go to state 6
WARNING: 
WARNING: Conflicts:
WARNING: 
WARNING: shift/reduce conflict for ARROW in state 11 resolved as shift
WARNING: shift/reduce conflict for CONJ in state 11 resolved as shift
WARNING: shift/reduce conflict for DISJ in state 11 resolved as shift
WARNING: shift/reduce conflict for NOT in state 11 resolved as shift
WARNING: reduce/reduce conflict in state 3 resolved using rule (ExpressionList -> <empty>)
WARNING: rejected rule (ExpressionList -> Expression) in state 3
WARNING: Rule (ExpressionList -> Expression) is never reduced
