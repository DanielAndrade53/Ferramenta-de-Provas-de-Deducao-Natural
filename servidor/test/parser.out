Created by PLY version 3.11 (http://www.dabeaz.com/ply)

Grammar

Rule 0     S' -> Program
Rule 1     Program -> ExpressionList
Rule 2     ExpressionList -> Expression
Rule 3     ExpressionList -> Expression ExpressionList
Rule 4     Expression -> ID
Rule 5     Expression -> UnaryOp ID
Rule 6     Expression -> Expression BinaryOp Expression
Rule 7     Expression -> UnaryOp LPAREN Expression RPAREN
Rule 8     Expression -> LPAREN Expression RPAREN
Rule 9     Program -> EExpressionList
Rule 10    EExpressionList -> EExpression
Rule 11    EExpressionList -> EExpression EExpressionList
Rule 12    EExpression -> EVar LPAREN ID RPAREN
Rule 13    EExpression -> EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN
Rule 14    EExpression -> EUnOp LPAREN UnaryOp COMMA EExpression RPAREN
Rule 15    BinaryOp -> ARROW
Rule 16    BinaryOp -> CONJ
Rule 17    BinaryOp -> DISJ
Rule 18    UnaryOp -> NOT

Terminals, with rules where they appear

ARROW                : 15
COMMA                : 13 13 14
CONJ                 : 16
DISJ                 : 17
EBinOp               : 13
EUnOp                : 14
EVar                 : 12
ID                   : 4 5 12
LPAREN               : 7 8 12 13 14
NOT                  : 18
RPAREN               : 7 8 12 13 14
error                : 

Nonterminals, with rules where they appear

BinaryOp             : 6 13
EExpression          : 10 11 13 13 14
EExpressionList      : 9 11
Expression           : 2 3 6 6 7 8
ExpressionList       : 1 3
Program              : 0
UnaryOp              : 5 7 14

Parsing method: LALR

state 0

    (0) S' -> . Program
    (1) Program -> . ExpressionList
    (9) Program -> . EExpressionList
    (2) ExpressionList -> . Expression
    (3) ExpressionList -> . Expression ExpressionList
    (10) EExpressionList -> . EExpression
    (11) EExpressionList -> . EExpression EExpressionList
    (4) Expression -> . ID
    (5) Expression -> . UnaryOp ID
    (6) Expression -> . Expression BinaryOp Expression
    (7) Expression -> . UnaryOp LPAREN Expression RPAREN
    (8) Expression -> . LPAREN Expression RPAREN
    (12) EExpression -> . EVar LPAREN ID RPAREN
    (13) EExpression -> . EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN
    (14) EExpression -> . EUnOp LPAREN UnaryOp COMMA EExpression RPAREN
    (18) UnaryOp -> . NOT

    ID              shift and go to state 6
    LPAREN          shift and go to state 8
    EVar            shift and go to state 9
    EBinOp          shift and go to state 10
    EUnOp           shift and go to state 11
    NOT             shift and go to state 12

    Program                        shift and go to state 1
    ExpressionList                 shift and go to state 2
    EExpressionList                shift and go to state 3
    Expression                     shift and go to state 4
    EExpression                    shift and go to state 5
    UnaryOp                        shift and go to state 7

state 1

    (0) S' -> Program .



state 2

    (1) Program -> ExpressionList .

    $end            reduce using rule 1 (Program -> ExpressionList .)


state 3

    (9) Program -> EExpressionList .

    $end            reduce using rule 9 (Program -> EExpressionList .)


state 4

    (2) ExpressionList -> Expression .
    (3) ExpressionList -> Expression . ExpressionList
    (6) Expression -> Expression . BinaryOp Expression
    (2) ExpressionList -> . Expression
    (3) ExpressionList -> . Expression ExpressionList
    (15) BinaryOp -> . ARROW
    (16) BinaryOp -> . CONJ
    (17) BinaryOp -> . DISJ
    (4) Expression -> . ID
    (5) Expression -> . UnaryOp ID
    (6) Expression -> . Expression BinaryOp Expression
    (7) Expression -> . UnaryOp LPAREN Expression RPAREN
    (8) Expression -> . LPAREN Expression RPAREN
    (18) UnaryOp -> . NOT

    $end            reduce using rule 2 (ExpressionList -> Expression .)
    ARROW           shift and go to state 15
    CONJ            shift and go to state 16
    DISJ            shift and go to state 17
    ID              shift and go to state 6
    LPAREN          shift and go to state 8
    NOT             shift and go to state 12

    Expression                     shift and go to state 4
    ExpressionList                 shift and go to state 13
    BinaryOp                       shift and go to state 14
    UnaryOp                        shift and go to state 7

state 5

    (10) EExpressionList -> EExpression .
    (11) EExpressionList -> EExpression . EExpressionList
    (10) EExpressionList -> . EExpression
    (11) EExpressionList -> . EExpression EExpressionList
    (12) EExpression -> . EVar LPAREN ID RPAREN
    (13) EExpression -> . EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN
    (14) EExpression -> . EUnOp LPAREN UnaryOp COMMA EExpression RPAREN

    $end            reduce using rule 10 (EExpressionList -> EExpression .)
    EVar            shift and go to state 9
    EBinOp          shift and go to state 10
    EUnOp           shift and go to state 11

    EExpression                    shift and go to state 5
    EExpressionList                shift and go to state 18

state 6

    (4) Expression -> ID .

    ARROW           reduce using rule 4 (Expression -> ID .)
    CONJ            reduce using rule 4 (Expression -> ID .)
    DISJ            reduce using rule 4 (Expression -> ID .)
    ID              reduce using rule 4 (Expression -> ID .)
    LPAREN          reduce using rule 4 (Expression -> ID .)
    NOT             reduce using rule 4 (Expression -> ID .)
    $end            reduce using rule 4 (Expression -> ID .)
    RPAREN          reduce using rule 4 (Expression -> ID .)


state 7

    (5) Expression -> UnaryOp . ID
    (7) Expression -> UnaryOp . LPAREN Expression RPAREN

    ID              shift and go to state 19
    LPAREN          shift and go to state 20


state 8

    (8) Expression -> LPAREN . Expression RPAREN
    (4) Expression -> . ID
    (5) Expression -> . UnaryOp ID
    (6) Expression -> . Expression BinaryOp Expression
    (7) Expression -> . UnaryOp LPAREN Expression RPAREN
    (8) Expression -> . LPAREN Expression RPAREN
    (18) UnaryOp -> . NOT

    ID              shift and go to state 6
    LPAREN          shift and go to state 8
    NOT             shift and go to state 12

    Expression                     shift and go to state 21
    UnaryOp                        shift and go to state 7

state 9

    (12) EExpression -> EVar . LPAREN ID RPAREN

    LPAREN          shift and go to state 22


state 10

    (13) EExpression -> EBinOp . LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN

    LPAREN          shift and go to state 23


state 11

    (14) EExpression -> EUnOp . LPAREN UnaryOp COMMA EExpression RPAREN

    LPAREN          shift and go to state 24


state 12

    (18) UnaryOp -> NOT .

    ID              reduce using rule 18 (UnaryOp -> NOT .)
    LPAREN          reduce using rule 18 (UnaryOp -> NOT .)
    COMMA           reduce using rule 18 (UnaryOp -> NOT .)


state 13

    (3) ExpressionList -> Expression ExpressionList .

    $end            reduce using rule 3 (ExpressionList -> Expression ExpressionList .)


state 14

    (6) Expression -> Expression BinaryOp . Expression
    (4) Expression -> . ID
    (5) Expression -> . UnaryOp ID
    (6) Expression -> . Expression BinaryOp Expression
    (7) Expression -> . UnaryOp LPAREN Expression RPAREN
    (8) Expression -> . LPAREN Expression RPAREN
    (18) UnaryOp -> . NOT

    ID              shift and go to state 6
    LPAREN          shift and go to state 8
    NOT             shift and go to state 12

    Expression                     shift and go to state 25
    UnaryOp                        shift and go to state 7

state 15

    (15) BinaryOp -> ARROW .

    ID              reduce using rule 15 (BinaryOp -> ARROW .)
    LPAREN          reduce using rule 15 (BinaryOp -> ARROW .)
    NOT             reduce using rule 15 (BinaryOp -> ARROW .)
    COMMA           reduce using rule 15 (BinaryOp -> ARROW .)


state 16

    (16) BinaryOp -> CONJ .

    ID              reduce using rule 16 (BinaryOp -> CONJ .)
    LPAREN          reduce using rule 16 (BinaryOp -> CONJ .)
    NOT             reduce using rule 16 (BinaryOp -> CONJ .)
    COMMA           reduce using rule 16 (BinaryOp -> CONJ .)


state 17

    (17) BinaryOp -> DISJ .

    ID              reduce using rule 17 (BinaryOp -> DISJ .)
    LPAREN          reduce using rule 17 (BinaryOp -> DISJ .)
    NOT             reduce using rule 17 (BinaryOp -> DISJ .)
    COMMA           reduce using rule 17 (BinaryOp -> DISJ .)


state 18

    (11) EExpressionList -> EExpression EExpressionList .

    $end            reduce using rule 11 (EExpressionList -> EExpression EExpressionList .)


state 19

    (5) Expression -> UnaryOp ID .

    ARROW           reduce using rule 5 (Expression -> UnaryOp ID .)
    CONJ            reduce using rule 5 (Expression -> UnaryOp ID .)
    DISJ            reduce using rule 5 (Expression -> UnaryOp ID .)
    ID              reduce using rule 5 (Expression -> UnaryOp ID .)
    LPAREN          reduce using rule 5 (Expression -> UnaryOp ID .)
    NOT             reduce using rule 5 (Expression -> UnaryOp ID .)
    $end            reduce using rule 5 (Expression -> UnaryOp ID .)
    RPAREN          reduce using rule 5 (Expression -> UnaryOp ID .)


state 20

    (7) Expression -> UnaryOp LPAREN . Expression RPAREN
    (4) Expression -> . ID
    (5) Expression -> . UnaryOp ID
    (6) Expression -> . Expression BinaryOp Expression
    (7) Expression -> . UnaryOp LPAREN Expression RPAREN
    (8) Expression -> . LPAREN Expression RPAREN
    (18) UnaryOp -> . NOT

    ID              shift and go to state 6
    LPAREN          shift and go to state 8
    NOT             shift and go to state 12

    UnaryOp                        shift and go to state 7
    Expression                     shift and go to state 26

state 21

    (8) Expression -> LPAREN Expression . RPAREN
    (6) Expression -> Expression . BinaryOp Expression
    (15) BinaryOp -> . ARROW
    (16) BinaryOp -> . CONJ
    (17) BinaryOp -> . DISJ

    RPAREN          shift and go to state 27
    ARROW           shift and go to state 15
    CONJ            shift and go to state 16
    DISJ            shift and go to state 17

    BinaryOp                       shift and go to state 14

state 22

    (12) EExpression -> EVar LPAREN . ID RPAREN

    ID              shift and go to state 28


state 23

    (13) EExpression -> EBinOp LPAREN . BinaryOp COMMA EExpression COMMA EExpression RPAREN
    (15) BinaryOp -> . ARROW
    (16) BinaryOp -> . CONJ
    (17) BinaryOp -> . DISJ

    ARROW           shift and go to state 15
    CONJ            shift and go to state 16
    DISJ            shift and go to state 17

    BinaryOp                       shift and go to state 29

state 24

    (14) EExpression -> EUnOp LPAREN . UnaryOp COMMA EExpression RPAREN
    (18) UnaryOp -> . NOT

    NOT             shift and go to state 12

    UnaryOp                        shift and go to state 30

state 25

    (6) Expression -> Expression BinaryOp Expression .
    (6) Expression -> Expression . BinaryOp Expression
    (15) BinaryOp -> . ARROW
    (16) BinaryOp -> . CONJ
    (17) BinaryOp -> . DISJ

  ! shift/reduce conflict for ARROW resolved as shift
  ! shift/reduce conflict for CONJ resolved as shift
  ! shift/reduce conflict for DISJ resolved as shift
    ID              reduce using rule 6 (Expression -> Expression BinaryOp Expression .)
    LPAREN          reduce using rule 6 (Expression -> Expression BinaryOp Expression .)
    NOT             reduce using rule 6 (Expression -> Expression BinaryOp Expression .)
    $end            reduce using rule 6 (Expression -> Expression BinaryOp Expression .)
    RPAREN          reduce using rule 6 (Expression -> Expression BinaryOp Expression .)
    ARROW           shift and go to state 15
    CONJ            shift and go to state 16
    DISJ            shift and go to state 17

  ! ARROW           [ reduce using rule 6 (Expression -> Expression BinaryOp Expression .) ]
  ! CONJ            [ reduce using rule 6 (Expression -> Expression BinaryOp Expression .) ]
  ! DISJ            [ reduce using rule 6 (Expression -> Expression BinaryOp Expression .) ]

    BinaryOp                       shift and go to state 14

state 26

    (7) Expression -> UnaryOp LPAREN Expression . RPAREN
    (6) Expression -> Expression . BinaryOp Expression
    (15) BinaryOp -> . ARROW
    (16) BinaryOp -> . CONJ
    (17) BinaryOp -> . DISJ

    RPAREN          shift and go to state 31
    ARROW           shift and go to state 15
    CONJ            shift and go to state 16
    DISJ            shift and go to state 17

    BinaryOp                       shift and go to state 14

state 27

    (8) Expression -> LPAREN Expression RPAREN .

    ARROW           reduce using rule 8 (Expression -> LPAREN Expression RPAREN .)
    CONJ            reduce using rule 8 (Expression -> LPAREN Expression RPAREN .)
    DISJ            reduce using rule 8 (Expression -> LPAREN Expression RPAREN .)
    ID              reduce using rule 8 (Expression -> LPAREN Expression RPAREN .)
    LPAREN          reduce using rule 8 (Expression -> LPAREN Expression RPAREN .)
    NOT             reduce using rule 8 (Expression -> LPAREN Expression RPAREN .)
    $end            reduce using rule 8 (Expression -> LPAREN Expression RPAREN .)
    RPAREN          reduce using rule 8 (Expression -> LPAREN Expression RPAREN .)


state 28

    (12) EExpression -> EVar LPAREN ID . RPAREN

    RPAREN          shift and go to state 32


state 29

    (13) EExpression -> EBinOp LPAREN BinaryOp . COMMA EExpression COMMA EExpression RPAREN

    COMMA           shift and go to state 33


state 30

    (14) EExpression -> EUnOp LPAREN UnaryOp . COMMA EExpression RPAREN

    COMMA           shift and go to state 34


state 31

    (7) Expression -> UnaryOp LPAREN Expression RPAREN .

    ARROW           reduce using rule 7 (Expression -> UnaryOp LPAREN Expression RPAREN .)
    CONJ            reduce using rule 7 (Expression -> UnaryOp LPAREN Expression RPAREN .)
    DISJ            reduce using rule 7 (Expression -> UnaryOp LPAREN Expression RPAREN .)
    ID              reduce using rule 7 (Expression -> UnaryOp LPAREN Expression RPAREN .)
    LPAREN          reduce using rule 7 (Expression -> UnaryOp LPAREN Expression RPAREN .)
    NOT             reduce using rule 7 (Expression -> UnaryOp LPAREN Expression RPAREN .)
    $end            reduce using rule 7 (Expression -> UnaryOp LPAREN Expression RPAREN .)
    RPAREN          reduce using rule 7 (Expression -> UnaryOp LPAREN Expression RPAREN .)


state 32

    (12) EExpression -> EVar LPAREN ID RPAREN .

    EVar            reduce using rule 12 (EExpression -> EVar LPAREN ID RPAREN .)
    EBinOp          reduce using rule 12 (EExpression -> EVar LPAREN ID RPAREN .)
    EUnOp           reduce using rule 12 (EExpression -> EVar LPAREN ID RPAREN .)
    $end            reduce using rule 12 (EExpression -> EVar LPAREN ID RPAREN .)
    COMMA           reduce using rule 12 (EExpression -> EVar LPAREN ID RPAREN .)
    RPAREN          reduce using rule 12 (EExpression -> EVar LPAREN ID RPAREN .)


state 33

    (13) EExpression -> EBinOp LPAREN BinaryOp COMMA . EExpression COMMA EExpression RPAREN
    (12) EExpression -> . EVar LPAREN ID RPAREN
    (13) EExpression -> . EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN
    (14) EExpression -> . EUnOp LPAREN UnaryOp COMMA EExpression RPAREN

    EVar            shift and go to state 9
    EBinOp          shift and go to state 10
    EUnOp           shift and go to state 11

    EExpression                    shift and go to state 35

state 34

    (14) EExpression -> EUnOp LPAREN UnaryOp COMMA . EExpression RPAREN
    (12) EExpression -> . EVar LPAREN ID RPAREN
    (13) EExpression -> . EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN
    (14) EExpression -> . EUnOp LPAREN UnaryOp COMMA EExpression RPAREN

    EVar            shift and go to state 9
    EBinOp          shift and go to state 10
    EUnOp           shift and go to state 11

    EExpression                    shift and go to state 36

state 35

    (13) EExpression -> EBinOp LPAREN BinaryOp COMMA EExpression . COMMA EExpression RPAREN

    COMMA           shift and go to state 37


state 36

    (14) EExpression -> EUnOp LPAREN UnaryOp COMMA EExpression . RPAREN

    RPAREN          shift and go to state 38


state 37

    (13) EExpression -> EBinOp LPAREN BinaryOp COMMA EExpression COMMA . EExpression RPAREN
    (12) EExpression -> . EVar LPAREN ID RPAREN
    (13) EExpression -> . EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN
    (14) EExpression -> . EUnOp LPAREN UnaryOp COMMA EExpression RPAREN

    EVar            shift and go to state 9
    EBinOp          shift and go to state 10
    EUnOp           shift and go to state 11

    EExpression                    shift and go to state 39

state 38

    (14) EExpression -> EUnOp LPAREN UnaryOp COMMA EExpression RPAREN .

    EVar            reduce using rule 14 (EExpression -> EUnOp LPAREN UnaryOp COMMA EExpression RPAREN .)
    EBinOp          reduce using rule 14 (EExpression -> EUnOp LPAREN UnaryOp COMMA EExpression RPAREN .)
    EUnOp           reduce using rule 14 (EExpression -> EUnOp LPAREN UnaryOp COMMA EExpression RPAREN .)
    $end            reduce using rule 14 (EExpression -> EUnOp LPAREN UnaryOp COMMA EExpression RPAREN .)
    COMMA           reduce using rule 14 (EExpression -> EUnOp LPAREN UnaryOp COMMA EExpression RPAREN .)
    RPAREN          reduce using rule 14 (EExpression -> EUnOp LPAREN UnaryOp COMMA EExpression RPAREN .)


state 39

    (13) EExpression -> EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression . RPAREN

    RPAREN          shift and go to state 40


state 40

    (13) EExpression -> EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN .

    EVar            reduce using rule 13 (EExpression -> EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN .)
    EBinOp          reduce using rule 13 (EExpression -> EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN .)
    EUnOp           reduce using rule 13 (EExpression -> EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN .)
    $end            reduce using rule 13 (EExpression -> EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN .)
    COMMA           reduce using rule 13 (EExpression -> EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN .)
    RPAREN          reduce using rule 13 (EExpression -> EBinOp LPAREN BinaryOp COMMA EExpression COMMA EExpression RPAREN .)

WARNING: 
WARNING: Conflicts:
WARNING: 
WARNING: shift/reduce conflict for ARROW in state 25 resolved as shift
WARNING: shift/reduce conflict for CONJ in state 25 resolved as shift
WARNING: shift/reduce conflict for DISJ in state 25 resolved as shift
